robusta_krr/core/runner.py

```diff
@@ -116,7 +121,8 @@ async def _calculate_object_recommendations(self, object: K8sObjectData) -> tupl
 
         # NOTE: We run this in a threadpool as the strategy calculation might be CPU intensive
         # But keep in mind that numpy calcluations will not block the GIL
-        result = await asyncio.to_thread(self._strategy.run, data, object)
+        loop = asyncio.get_running_loop()
+        result = await loop.run_in_executor(self._executor, self._strategy.run, data, object)
```

Comment:
do we want to implement some sort of rate limiter here also, there will still could be almost constant queries to prometheus

Reply:
Same, we limited the query rate by limiting the executor